### ISSUE TITLE
[BUG] Day component not submittable if required on all set to true

### ISSUE THREAD

**Issue by aminmc** [formio/formio.js#1821]
## Environment
Please provide as many details as you can:

- Hosting type
  - [x] Form.io
  - [ ] Local deployment
    - Version: 
- Formio.js version: 4.2.12
- Frontend framework: react-formio
- Browser:
- Browser version:

The following provides a schema for a wizard with a day component.  
```json
{
  "name": "testWizardDebugDayComponent",
  "path": "test-wizard-debug-day-component",
  "title": "test-wizard-debug-day-component",
  "display": "wizard",
  "components": [
    {
      "id": "e8zez88",
      "key": "page1",
      "path": "page1",
      "tree": false,
      "type": "panel",
      "input": false,
      "label": "Page 1",
      "theme": "default",
      "title": "Page 1",
      "hidden": false,
      "prefix": "",
      "suffix": "",
      "unique": false,
      "widget": null,
      "dbIndex": false,
      "overlay": {
        "top": "",
        "left": "",
        "style": "",
        "width": "",
        "height": ""
      },
      "tooltip": "",
      "disabled": false,
      "multiple": false,
      "redrawOn": "",
      "tabindex": "",
      "validate": {
        "custom": "",
        "required": false,
        "customPrivate": false,
        "strictDateValidation": false
      },
      "autofocus": false,
      "encrypted": false,
      "hideLabel": false,
      "protected": false,
      "refreshOn": "",
      "tableView": false,
      "attributes": {},
      "breadcrumb": "default",
      "components": [
        {
          "id": "ey8xf8m",
          "key": "expiryDate",
          "tags": "",
          "type": "day",
          "input": true,
          "label": "Expiry date",
          "logic": [],
          "fields": {
            "day": {
              "hide": false,
              "type": "number",
              "required": true,
              "placeholder": ""
            },
            "year": {
              "hide": false,
              "type": "number",
              "required": true,
              "placeholder": ""
            },
            "month": {
              "hide": false,
              "type": "select",
              "required": true,
              "placeholder": ""
            }
          },
          "hidden": false,
          "prefix": "",
          "suffix": "",
          "unique": false,
          "widget": null,
          "dbIndex": false,
          "maxDate": "",
          "minDate": "",
          "overlay": {
            "top": "",
            "left": "",
            "page": "",
            "style": "",
            "width": "",
            "height": ""
          },
          "tooltip": "",
          "dayFirst": true,
          "disabled": false,
          "multiple": false,
          "redrawOn": "",
          "tabindex": "",
          "validate": {
            "json": "",
            "custom": "",
            "required": false,
            "customMessage": "",
            "customPrivate": false,
            "strictDateValidation": false
          },
          "autofocus": false,
          "encrypted": false,
          "hideLabel": false,
          "protected": false,
          "refreshOn": "",
          "tableView": false,
          "attributes": {},
          "errorLabel": "",
          "persistent": true,
          "properties": {},
          "validateOn": "change",
          "clearOnHide": true,
          "conditional": {
            "eq": "",
            "json": "",
            "show": null,
            "when": null
          },
          "customClass": "",
          "description": "",
          "placeholder": "",
          "defaultValue": "",
          "alwaysEnabled": false,
          "labelPosition": "top",
          "showCharCount": false,
          "showWordCount": false,
          "calculateValue": "",
          "hideInputLabels": false,
          "customConditional": "",
          "useLocaleSettings": false,
          "allowMultipleMasks": false,
          "customDefaultValue": "",
          "inputsLabelPosition": "top",
          "allowCalculateOverride": false
        }
      ],
      "errorLabel": "",
      "persistent": false,
      "properties": {},
      "validateOn": "change",
      "clearOnHide": false,
      "conditional": {
        "eq": "",
        "show": null,
        "when": null
      },
      "customClass": "",
      "description": "",
      "placeholder": "",
      "defaultValue": null,
      "alwaysEnabled": false,
      "labelPosition": "top",
      "showCharCount": false,
      "showWordCount": false,
      "calculateValue": "",
      "allowMultipleMasks": false,
      "customDefaultValue": "",
      "allowCalculateOverride": false
    },
    {
      "id": "ek8g3rm",
      "key": "page2",
      "path": "page2",
      "tree": false,
      "type": "panel",
      "input": false,
      "label": "Page 2",
      "theme": "default",
      "title": "Page 2",
      "hidden": false,
      "prefix": "",
      "suffix": "",
      "unique": false,
      "widget": null,
      "dbIndex": false,
      "overlay": {
        "top": "",
        "left": "",
        "style": "",
        "width": "",
        "height": ""
      },
      "tooltip": "",
      "disabled": false,
      "multiple": false,
      "redrawOn": "",
      "tabindex": "",
      "validate": {
        "custom": "",
        "required": false,
        "customPrivate": false
      },
      "autofocus": false,
      "encrypted": false,
      "hideLabel": false,
      "protected": false,
      "refreshOn": "",
      "tableView": false,
      "attributes": {},
      "breadcrumb": "default",
      "components": [],
      "errorLabel": "",
      "persistent": false,
      "properties": {},
      "validateOn": "change",
      "clearOnHide": false,
      "conditional": {
        "eq": "",
        "show": null,
        "when": null
      },
      "customClass": "",
      "description": "",
      "placeholder": "",
      "defaultValue": null,
      "alwaysEnabled": false,
      "labelPosition": "top",
      "showCharCount": false,
      "showWordCount": false,
      "calculateValue": "",
      "allowMultipleMasks": false,
      "customDefaultValue": "",
      "allowCalculateOverride": false
    }
  ]
}
```

If the component is set  day, month and year to be required (using checkbox) you cannot submit the form even if you enter the day, month and year.

The following error is caught:

![Screenshot 2019-10-07 at 14 05 49](https://user-images.githubusercontent.com/290639/66314604-33d53180-e90c-11e9-8980-22771ec90794.png)

If you apply conditional logic using valid !== '00/00/000' then it works.

**Comment by travist** [comment]
Looks like a bug, and at first glance, looks like a bad one. We will prioritize this to get fixed soon.

Thanks,

Travis.

**Comment by jhen1422** [comment]
@aminmc 
Thanks for the report. I was able to replicate the issue and have created a ticket. We will get this prioritized during our planning meeting this morning.

Thanks!

**Comment by aminmc** [comment]
@jhen1422 @travist this looks resolved against 4.3.3
